### PR TITLE
PathArgument for separating between string and path arguments

### PR DESCRIPTION
**Pull request by Bouke**
This PR is split off from #843 per Ankit's request. It lays the ground-work for #843.

### CODE REVIEW

File: Sources/TestSupportExecutable/main.swift [hunk 1]
```diff
@@ -48,6 +48,7 @@ class HandlerTest {
 enum Mode: String {
     case fileLockTest
     case interruptHandlerTest
+    case absolutePath
```
Comment (aciidb0mb3r) [inline Sources/TestSupportExecutable/main.swift]: I would call this `pathArgumentTest`

File: Sources/Utility/ArgumentParser.swift [hunk 2]
```diff
@@ -113,6 +113,21 @@ extension StringEnumArgument {
...
+public struct PathArgument: ArgumentKind {
+    public let path: AbsolutePath
+
+    public init?(arg: String) {
+        path = AbsolutePath(arg, relativeTo: currentWorkingDirectory)
+    }
+
+    public init(parser: inout ArgumentParserProtocol) throws {
+        path = AbsolutePath(try parser.associatedArgumentValue ?? parser.next(), relativeTo: currentWorkingDirectory)
+    }
```
Comment (aciidb0mb3r) [inline Sources/Utility/ArgumentParser.swift]: This will fail if arguments are not representable as relative paths. We should ensure validity of the path before passing it to `AbsolutePath`

File: Tests/UtilityTests/ArgumentParserTests.swift [hunk 3]
```diff
@@ -367,6 +368,21 @@ class ArgumentParserTests: XCTestCase {
         }
     }
 
+    func testPathArgument() {
+        // relative path
```
Comment (aciidb0mb3r) [inline Tests/UtilityTests/ArgumentParserTests.swift]: better comments appreciated! foe e.g.: `// Test that relative path is resolved.`

File: Tests/UtilityTests/ArgumentParserTests.swift [hunk 4]
```diff
@@ -367,6 +368,21 @@ class ArgumentParserTests: XCTestCase {
...
+        do {
+            let actual = try! SwiftPMProduct.TestSupportExecutable.execute(["absolutePath", "some/path"]).chomp()
+            let expected = currentWorkingDirectory.appending(RelativePath("some/path")).asString
+            XCTAssertEqual(actual, expected)
+        }
+
+        // absolute path
+        do {
+            let actual = try! SwiftPMProduct.TestSupportExecutable.execute(["absolutePath", "/bin/echo"]).chomp()
+            XCTAssertEqual(actual, "/bin/echo")
```
Comment (aciidb0mb3r) [inline Tests/UtilityTests/ArgumentParserTests.swift]: Lets add some failure case where user input is an invalid crash and make sure we display appropriate error/diagnosis.

Comment (Bouke) [inline Tests/UtilityTests/ArgumentParserTests.swift]: You're probably better informed about AbsolulePath than I am, so do you have some suggestions for such failure cases?

Comment (aciidb0mb3r) [inline Tests/UtilityTests/ArgumentParserTests.swift]: Actually this can accept both paths so there should be no failures. But we should still add some edge cases for eg if `../foo` and `./foo` resolves correctly etc.

File: Sources/Utility/ArgumentParser.swift (hunk 2 above)
Comment (Bouke) [inline Sources/Utility/ArgumentParser.swift]: Also absolute paths are valid arguments, so should be no problem I suppose.

**Comment by aciidb0mb3r** [comment]
@swift-ci please test and merge
